rebasehelper/tests/test_specfile.py

```diff
@@ -177,23 +243,50 @@ def test_set_extra_version(self, spec_object):
         spec_object.set_extra_version('g1234567', False)
         assert spec_object.get_release() == '2.g1234567'
 
-    def test_update_setup_dirname(self, spec_object):
-        spec_object.set_extra_version('rc1', False)
+    @pytest.mark.parametrize('spec_attributes', [
+        {
+            'spec_content': dedent("""\
+                %global release 34
+                %global release_str %{release}%{?dist}
 
-        prep = spec_object.spec_content.section('%prep')
-        spec_object.update_setup_dirname('test-1.0.2')
-        assert spec_object.spec_content.section('%prep') == prep
+                Name:    test
+                Version: 1.0.2
+                Release: %{release_str}
 
-        spec_object.update_setup_dirname('test-1.0.2rc1')
-        prep = spec_object.spec_content.section('%prep')
+                %prep
+                %setup -q -c -a 5
+                """),
+            'get_release': lambda: '34',
+        }
+    ])
+    def test_update_setup_dirname(self, mocked_spec_object):
+        # mock %{version} and %{name} macros, set their level to -3
+        macros = {'name': 'test', 'version': '1.0.2'}
+        for macro, value in macros.items():
+            rpm.addMacro(macro, value)
+        mocked_spec_object.macros = MacroHelper.dump()
+        for m in macros:
+            for macro in mocked_spec_object.macros:
+                if macro['name'] == m:
+                    macro['level'] = -3
```

Comment:
This definitely shouldn't be part of a test. `mocked_spec_object` fixture should instead accept another argument like `macros`.

Reply:
Or maybe, to stay consistent, it should process `macros` key in `spec_attributes` analogically to how `spec_content` is being processed.